pyccel/parser/semantic.py

```diff
@@ -1392,10 +1392,14 @@ def _assign_GeneratorComprehension(self, lhs_name, expr, **settings):
         # Iterate over the loops
         # This provides the definitions of iterators as well
         # as the central expression
-        loops  = [self._visit(expr.loops, **settings)]
 
         # If necessary add additional expressions corresponding
         # to nested GeneratorComprehensions
+        loops = [self._visit(expr.loops, **settings)]
+        add_exprs  = [*self._additional_exprs[-1]]
+        loops = add_exprs + loops
+        self._additional_exprs.pop()
+        self._additional_exprs.append([])
```

Comment:
`self._additional_exprs[-1]` is already a list so upacking it into a list is pointless.
```python
add_exprs  = [*self._additional_exprs[-1]]
```
is equivalent to:
```python
add_exprs  = list(self._additional_exprs[-1])
```
or
```python
add_exprs  = self._additional_exprs[-1].copy()
```
But there is no need to make a copy here.

Also `pop()` returns the element that it is removing so this section could be written much more compactly as:
```python
loops = self._additional_exprs.pop() + loops
self._additional_exprs.append([])
```

Reply:
Sure that looks cleaner, I'll fix that.

Reply:
I ended up dropping the insertion in the loop list in [b26a937](https://github.com/pyccel/pyccel/pull/1187/commits/b26a937f05f32cee939135004c3977b37f9b462f).